modules/xcursor.nix

```diff
@@ -0,0 +1,75 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.xcursor;
+
+  cursorType = types.submodule {
+    options = {
+      package = mkOption {
+        type = types.package;
+        example = literalExample "pkgs.vanilla-dmz";
+        description = "Package providing the cursor theme.";
+      };
+
+      name = mkOption {
+        type = types.str;
+        example = "Vanilla-DMZ";
+        description = "The cursor name within the package.";
+      };
+
+      size = mkOption {
+        type = types.int;
+        default = 32;
+        example = 64;
+        description = "The cursor size.";
+      };
+    };
+  };
+
+in
+
+{
+  meta.maintainers = [ maintainers.league ];
+
+  options = {
+    xcursor = mkOption {
```

Comment:
I'd suggest the name `xsession.cursor` for this option.

Reply:
And yeah, I think my choice of name for the `xresources` option was flawed. I think it should be called something like `xsession.resourceProperties` instead 😄

Reply:
Actually, reading the xsetroot man page it talks about "pointer cursor" so the option name `xsession.pointerCursor` might be more descriptive.